src/main/java/Entities/Player.java

```diff
@@ -0,0 +1,38 @@
+package Entities;
+import java.util.HashMap;
+
+public class Player {
+
+	public String username;
+	private HashMap<String, GameStats> playerGames;
+	
+	public Player(String username) {
+		this.username = username;
+		this.playerGames = new HashMap<String, GameStats>();
+	}
+
+	public void registerFor(String sport) {
+		if (!playerGames.containsKey(sport)) {
+			playerGames.put(sport, new GameStats());
+		}
+	}
+	
+	public GameStats getPlayerGamesFor(String sport) {
+		int wins, loss, winstreak;
+		if (playerGames.containsKey(sport)) {
+			wins = playerGames.get(sport).getWins();
+			loss = playerGames.get(sport).getLoss();
+			winstreak = playerGames.get(sport).getWinstreak();
+			return new GameStats(wins, loss, winstreak);
+		}
+		return null;
+	}
+
+	public void updateGameStatsAfter_WinIn(String sport) {
```

Comment:
What are the advantages of storing records for a player in the player model vs in the game itself?

Reply:
You mean have it on all under each games? like so:
"ping pong": [player1, player2, player3],
"soccer": []
?
I was thinking:
```json
"pingpong": {
   "players": ["p1_username", "p2_username", "p3_username"], 
   "teams": [not-yet-explored], 
   "metadata": {"team-limit": 2}, 
   "houses": [not-yet-explored], 
   "brackets":["burning-sun", "lava", "blazing-fire", "smouldering-fire"]
    }
}
```
### Meanwhile in Players
```json
"ready_player_one@oracle.com": {
   "teams": {"pingpong":[]},
   "house": "TootyFrootie",
   "games": [{
        "pingpong":{"win":1, "loss":24, "winstreak": 1, 
               preferences: "rely-on-honor-system"}
    }],
   "achievement": ["played-all-games", "burning-man"]
}
```
- I guess Players would be for player dashboard and stats, badges, house, personal teams?
- For viewing leaderboards and brackets, we get list of usernames from game records, and query player records for displaying player data? vice versa for the above bullet. 
- We can also store teams in game? and have player details separated, just seems easier to handle/debug; simpler to maintain?
- Separation of concerns I guess. Players keep players and their extras; games keep games and their extras. I don't know why I thought of separating em when I first saw them. Am I thinking like it's relational?

*Please do correct me if there's a more effective way*

---

Or do you mean something else by ```storing it in the game itself```?

Reply:
That's not what we wrote though... I see, yeah scope is too small. Blind coding from the beginning. Will fix!